spec/models/bag_man_request_spec.rb

```diff
@@ -59,31 +59,42 @@
     end
 
     context "when not cancelled" do
+      let(:replication) {
+        mock = double(:replication_transfer)
+        allow(mock).to receive(:cancel!)
+        allow(mock).to receive(:from_node).and_return FromNode.new("fake_namespace")
+        mock
+      }
+      let(:request) {
+        request = Fabricate(:bag_man_request)
+        allow(request).to receive(:replication_transfer).and_return(replication)
+        request
+      }
       before(:each) do
-        @repl_mock = double(:replication_transfer)
-        allow(@repl_mock).to receive(:cancel!)
-        allow(@repl_mock).to receive(:from_node).and_return FromNode.new("fake_namespace")
-        @request = Fabricate(:bag_man_request)
-        allow(@request).to receive(:replication_transfer).and_return(@repl_mock)
         allow(Client::Repl::CancelJob).to receive(:perform_later)
       end
       it "sets the cancel_reason" do
-        @request.cancel!('testing')
-        expect(@request.reload.cancel_reason).to eql('testing')
+        request.cancel!('testing')
+        expect(request.reload.cancel_reason).to eql('testing')
       end
```

Comment:
I confirmed the above works as expected by changing it to `eql('what')` and that fails with
```
  1) BagManRequest#cancel! when not cancelled sets the cancel_reason
     Failure/Error:
       expect {
         request.cancel!('testing')
       }.to change {
         request.reload.cancel_reason
       }.to eql('what')

       expected result to have changed to eql "what", but is now "testing"
     # ./spec/models/bag_man_request_spec.rb:74:in `block (4 levels) in <top (required)>'
```